Implement Task Detailed View Redesign

jasperhuangg opened the pull request:
### Details
<!-- Explanation of the change or anything fishy that is going on -->

### Fixed Issues
<!---
1. Please replace GH_LINK with a URL link to the GitHub issue this Pull Request is fixing.
2. Please replace PROPOSAL: GH_LINK_ISSUE(COMMENT) with a URL link to your GitHub comment, which contains the approved proposal (i.e. the proposal that was approved by Expensify).

Do NOT add the special GH keywords like `fixed` etc, we have our own process of managing the flow.
It MUST be an entire link to the github issue and your comment proposal ; otherwise, the linking will not work as expected.

Make sure this section looks similar to this (you can link multiple issues using the same formatting, just add a new line):

$ https://github.com/Expensify/App/issues/<number-of-the-issue>
$ https://github.com/Expensify/App/issues/<number-of-the-issue(comment)>

Do NOT only link the issue number like this: $ #<number-of-the-issue>
--->
$ https://github.com/Expensify/App/issues/19330
$ https://github.com/Expensify/App/pull/20841

### Tests/Offline tests/QA
These tests should function exactly the same when offline, since we're setting optimistic data for everything
1. FAB > Assign task. Finish the flow for creating the task. Or just navigate to an existing task.
2. Verify the UI looks as expected (see screenshots).
3. Click on the title item. Verify it opens a title input. Change the title to be a really long name. Verify the task title expands to contains the name, and then has ellipses if the name is too long.
4. Click on the description item. Verify it opens a description input. Change the description to be a really long name. Verify the task title expands to contains the name, and then has ellipses if the name is too long.
5. Assign the task to a person you have a chat with already. Verify that the assignee updates. (NOTE: there is currently a bug on main for assigning tasks to people you don't have chats with, so we won't test that here).
6. Click the checkmark to complete the task. Verify the text in the button above changes to "Mark as incomplete". Uncheck the checkmark. Verify the text in the button above changes to "Mark as done".
7. Repeat the same behavior with the button and verify the checkmark and button respond accordingly.

### PR Author Checklist
<!--
This is a checklist for PR authors. Please make sure to complete all tasks and check them off once you do, or else your PR will not be merged!
-->

- [x] I linked the correct issue in the `### Fixed Issues` section above
- [x] I wrote clear testing steps that cover the changes made in this PR
    - [x] I added steps for local testing in the `Tests` section
    - [x] I added steps for the expected offline behavior in the `Offline steps` section
    - [x] I added steps for Staging and/or Production testing in the `QA steps` section
    - [x] I added steps to cover failure scenarios (i.e. verify an input displays the correct error message if the entered data is not correct)
    - [x] I turned off my network connection and tested it while offline to ensure it matches the expected behavior (i.e. verify the default avatar icon is displayed if app is offline)
    - [x] I tested this PR with a [High Traffic account](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#high-traffic-accounts) against the staging or production API to ensure there are no regressions (e.g. long loading states that impact usability).
- [x] I included screenshots or videos for tests on [all platforms](https://github.com/Expensify/App/blob/main/contributingGuides/CONTRIBUTING.md#make-sure-you-can-test-on-all-platforms)
- [x] I ran the tests on **all platforms** & verified they passed on:
    - [x] Android / native
    - [x] Android / Chrome
    - [x] iOS / native
    - [x] iOS / Safari
    - [x] MacOS / Chrome / Safari
    - [x] MacOS / Desktop
- [x] I verified there are no console errors (if there's a console error not related to the PR, report it or open an issue for it to be fixed)
- [x] I followed proper code patterns (see [Reviewing the code](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md#reviewing-the-code))
    - [x] I verified that any callback methods that were added or modified are named for what the method does and never what callback they handle (i.e. `toggleReport` and not `onIconClick`)
    - [x] I verified that comments were added to code that is not self explanatory
    - [x] I verified that any new or modified comments were clear, correct English, and explained "why" the code was doing something instead of only explaining "what" the code was doing.
    - [x] I verified any copy / text shown in the product is localized by adding it to `src/languages/*` files and using the [translation method](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60)
      - [x] If any non-english text was added/modified, I verified the translation was requested/reviewed in #expensify-open-source and it was approved by an internal Expensify engineer. Link to Slack message:
    - [x] I verified all numbers, amounts, dates and phone numbers shown in the product are using the [localization methods](https://github.com/Expensify/App/blob/4bd99402cebdf4d7394e0d1f260879ea238197eb/src/components/withLocalize.js#L60-L68)
    - [x] I verified any copy / text that was added to the app is grammatically correct in English. It adheres to proper capitalization guidelines (note: only the first word of header/labels should be capitalized), and is approved by marketing by adding the `Waiting for Copy` label for a copy review on the original GH to get the correct copy.
    - [x] I verified proper file naming conventions were followed for any new files or renamed files. All non-platform specific files are named after what they export and are not named "index.js". All platform-specific files are named for the platform the code supports as outlined in the README.
    - [x] I verified the JSDocs style guidelines (in [`STYLE.md`](https://github.com/Expensify/App/blob/main/contributingGuides/STYLE.md#jsdocs)) were followed
- [x] If a new code pattern is added I verified it was agreed to be used by multiple Expensify engineers
- [x] I followed the guidelines as stated in the [Review Guidelines](https://github.com/Expensify/App/blob/main/contributingGuides/PR_REVIEW_GUIDELINES.md)
- [x] I tested other components that can be impacted by my changes (i.e. if the PR modifies a shared library or component like `Avatar`, I verified the components using `Avatar` are working as expected)
- [x] I verified all code is DRY (the PR doesn't include any logic written more than once, with the exception of tests)
- [x] I verified any variables that can be defined as constants (ie. in CONST.js or at the top of the file that uses the constant) are defined as such
- [x] I verified that if a function's arguments changed that all usages have also been updated correctly
- [x] If a new component is created I verified that:
    - [x] A similar component doesn't exist in the codebase
    - [x] All props are defined accurately and each prop has a `/** comment above it */`
    - [x] The file is named correctly
    - [x] The component has a clear name that is non-ambiguous and the purpose of the component can be inferred from the name alone
    - [x] The only data being stored in the state is data necessary for rendering and nothing else
    - [x] For Class Components, any internal methods passed to components event handlers are bound to `this` properly so there are no scoping issues (i.e. for `onClick={this.submit}` the method `this.submit` should be bound to `this` in the constructor)
    - [x] Any internal methods bound to `this` are necessary to be bound (i.e. avoid `this.submit = this.submit.bind(this);` if `this.submit` is never passed to a component event handler like `onClick`)
    - [x] All JSX used for rendering exists in the render method
    - [x] The component has the minimum amount of code necessary for its purpose, and it is broken down into smaller components in order to separate concerns and functions
- [x] If any new file was added I verified that:
    - [x] The file has a description of what it does and/or why is needed at the top of the file if the code is not self explanatory
- [x] If a new CSS style is added I verified that:
    - [x] A similar style doesn't already exist
    - [x] The style can't be created with an existing [StyleUtils](https://github.com/Expensify/App/blob/main/src/styles/StyleUtils.js) function (i.e. `StyleUtils.getBackgroundAndBorderStyle(themeColors.componentBG)`)
- [x] If the PR modifies code that runs when editing or sending messages, I tested and verified there is no unexpected behavior for all supported markdown - URLs, single line code, code blocks, quotes, headings, bold, strikethrough, and italic.
- [x] If the PR modifies a generic component, I tested and verified that those changes do not break usages of that component in the rest of the App (i.e. if a shared library or component like `Avatar` is modified, I verified that `Avatar` is working as expected in all cases)
- [x] If the PR modifies a component related to any of the existing Storybook stories, I tested and verified all stories for that component are still working as expected.
- [x] If a new page is added, I verified it's using the `ScrollView` component to make it scrollable when more elements are added to the page.
- [x] If the `main` branch was merged into this PR after a review, I tested again and verified the outcome was still expected according to the `Test` steps.
- [x] I have checked off every checkbox in the PR author checklist, including those that don't apply to this PR.

### Screenshots/Videos
<details>
<summary>Web</summary>

<img width="1726" alt="Screenshot 2023-06-22 at 6 08 52 PM" src="https://github.com/Expensify/App/assets/31285285/7bc264e8-3892-4128-9f88-86b3042bddfe">

</details>



<details>
<summary>Mobile Web - Chrome</summary>


<img width="390" alt="Screenshot 2023-06-22 at 8 07 31 PM" src="https://github.com/Expensify/App/assets/31285285/bce2cc5a-717c-49d9-bb81-ea733366c771">

</details>

<details>
<summary>Mobile Web - Safari</summary>


<img width="444" alt="Screenshot 2023-06-22 at 8 07 10 PM" src="https://github.com/Expensify/App/assets/31285285/1ae00366-b704-45af-bc09-3288426effb0">

</details>

<details>
<summary>Desktop</summary>

<img width="1201" alt="Screenshot 2023-06-22 at 6 11 12 PM" src="https://github.com/Expensify/App/assets/31285285/05c8ac9f-6a37-4fe2-bf2b-021686539493">


</details>

<details>
<summary>iOS</summary>


<img width="442" alt="Screenshot 2023-06-22 at 8 02 33 PM" src="https://github.com/Expensify/App/assets/31285285/6d994e16-e83c-46fa-95eb-3a271bd145ee">

</details>

<details>
<summary>Android</summary>

![Screenshot 2023-06-22 at 7 27 01 PM](https://github.com/Expensify/App/assets/31285285/ea675056-eee6-48af-b903-b969b501b8cc)


</details>

jasperhuangg committed:
component skeleton, render component in ReportActionList, show GBR on open task report LHN options

jasperhuangg committed:
remove gbr logic

jasperhuangg committed:
rename TaskUtils => Task, allow ReportActionItem to respond to report updates for task reports, implement title and description menu items in taskview

jasperhuangg committed:
add and use TaskHeaderActionButton

jasperhuangg committed:
fix translations

jasperhuangg committed:
remove unused

jasperhuangg committed:
remove unused

jasperhuangg committed:
merge main

jasperhuangg committed:
merge main

jasperhuangg committed:
style

jasperhuangg committed:
remove unused

jasperhuangg committed:
remove unused

jasperhuangg committed:
add created by and assignee to task view

jasperhuangg committed:
fix task preview assignee selection

jasperhuangg committed:
get correct avatar for each user

jasperhuangg committed:
description vertically centers if empty

thienlnam commented on src/components/LHNOptionsList/OptionRowLHN.js:
```diff
@@ -87,7 +87,7 @@ function OptionRowLHN(props) {
         !hasBrickError &&
         (optionItem.isUnreadWithMention ||
             (optionItem.hasOutstandingIOU && !optionItem.isIOUReportOwner) ||
-            (optionItem.isTaskReport && optionItem.isTaskAssignee && !optionItem.isTaskCompleted));
+            (optionItem.isTaskReport && optionItem.isTaskAssignee && !optionItem.isCompletedTaskReport));
```
Does this not work right now?

thienlnam replied:
Ah I see you updated the method

thienlnam commented on src/components/MenuItem.js:
```diff
@@ -97,6 +101,19 @@ function MenuItem(props) {
 
     const fallbackAvatarSize = props.viewMode === CONST.OPTION_MODE.COMPACT ? CONST.AVATAR_SIZE.SMALL : CONST.AVATAR_SIZE.DEFAULT;
 
+    const selectedState = props.shouldUseSquareSelectedState ? (
+        <Checkbox
+            onPress={() => props.onPressSelection()}
+            isChecked={props.isSelected}
+            style={props.shouldShowSelectedStateBeforeTitle && styles.mr3}
+        />
+    ) : (
+        <SelectCircle
+            isChecked={props.isSelected}
+            style={props.shouldShowSelectedStateBeforeTitle && styles.mr3}
+        />
+    );
```
When does this show a circle?

thienlnam replied:
Oh I see before this it would show the circle

thienlnam commented on src/libs/ReportUtils.js:
```diff
@@ -1067,7 +1091,7 @@ function getDMRootReportName(report) {
+    if (isChildReport(report)) {
         if (!getChatType(report)) {
             return `${Localize.translateLocal('threads.from')} ${getDMRootReportName(report)}`;
         }
```
No need to worry about the subtitle - will be taken care of in a different PR

thienlnam commented on src/components/TaskHeader.js:
```diff
@@ -1,123 +1,30 @@
-import React, {useEffect} from 'react';
```
I think we would remove this file completely actually, the only thing this is doing is showing the call to action button right? This will also be needed for manual requests, so maybe we can merge this into Header and make it a little more generalized

jasperhuangg replied:
done

jasperhuangg committed:
Merge branch 'main' of github.com:Expensify/App into jasper-taskRedesign

jasperhuangg committed:
lint

jasperhuangg committed:
fix ellipses

jasperhuangg committed:
move title input to its own component

jasperhuangg committed:
remove unused props

jasperhuangg committed:
whitespace

jasperhuangg committed:
style

jasperhuangg committed:
move styles into styles.js

jasperhuangg committed:
fix header button spacing on small screens

jasperhuangg committed:
remove unused import

jasperhuangg committed:
remove unused

jasperhuangg committed:
fix title menu item spacing

jasperhuangg committed:
fix title menu item spacing

jasperhuangg committed:
fix title menu item spacing

jasperhuangg committed:
fix title menu item spacing

jasperhuangg committed:
merge main

jasperhuangg committed:
merge main

jasperhuangg committed:
style

jasperhuangg committed:
Fix styling of task view (spacing)

jasperhuangg commented:
There's a bug with editing the assignee of an existing task to someone you've never spoken to before @thienlnam, but that actually already happens on main so I don't think we should block this PR on it.

thienlnam commented:
Yup agreed, makes sense to me

jasperhuangg committed:
fix task action styling on small screens

jasperhuangg committed:
fix title page display on android

jasperhuangg committed:
fix checkbox height

shawnborton commented:
For this:
<img width="415" alt="image" src="https://github.com/Expensify/App/assets/2319350/4a0b4490-016a-448c-a004-047e3fbc07a8">

Can we vertically align the checkbox with the title and arrow? So that it looks like this:
<img width="854" alt="image" src="https://github.com/Expensify/App/assets/2319350/a7193ac7-8d4a-4c82-95f2-3d7cd343dbf3">


For assignee:
<img width="410" alt="image" src="https://github.com/Expensify/App/assets/2319350/8fb2a4b9-69c6-4ae8-ae67-9ea51773c498">

...there should be 8px margin below the label Assignee and the avatar. The avatar should be 24px tall. 

We're also missing the empty state in the negative area above the task title:
<img width="682" alt="image" src="https://github.com/Expensify/App/assets/2319350/b5d4f584-a7a3-43fb-9ad4-f6311675a437">


For the horizontal rule style, we should only use that once a comment is added to the task. And then once it's added, it should not stretch full screen like you have it - it should have 20px padding on the left/right:
<img width="788" alt="image" src="https://github.com/Expensify/App/assets/2319350/c033a8c2-2640-41ea-8001-2d8f9553e84a">

jasperhuangg committed:
fix display for multiline titles

jasperhuangg commented:
@shawnborton 

I've fixed the vertical alignment. It's still not going to be 100% vertically aligned for lowercase letters since we need to leave space in the line height for uppercase letters, which for this particular font are a lot larger than lowercase letters.
<img width="504" alt="Screenshot 2023-06-23 at 10 57 25 AM" src="https://github.com/Expensify/App/assets/31285285/c54c5df9-95fb-44bb-bae7-96b0f6de3c6b">
<img width="495" alt="Screenshot 2023-06-23 at 10 58 34 AM" src="https://github.com/Expensify/App/assets/31285285/9d0d6698-1345-48bb-bb47-f7d3caf8246b">

I also fixed the assignee like you asked
<img width="302" alt="Screenshot 2023-06-23 at 11 32 58 AM" src="https://github.com/Expensify/App/assets/31285285/3b8c5d4b-f474-4be6-b589-d49c9e4ca160">

And the horizontal rule style
<img width="391" alt="Screenshot 2023-06-23 at 11 43 24 AM" src="https://github.com/Expensify/App/assets/31285285/3e9cbec4-b064-4486-8964-448104a10ea4">

> We're also missing the empty state in the negative area above the task title:

Can you clarify what you mean by this?

jasperhuangg committed:
address design review

jasperhuangg committed:
remove unused

jasperhuangg committed:
remove unused

shawnborton commented:
Rather than vertically center the checkbox and the text and the arrow, we might want the arrow and checkbox to align to the top, so it feels more like a bulleted list item:
<img width="337" alt="image" src="https://github.com/Expensify/App/assets/2319350/e8391911-9b02-478f-8743-24ff3f96f61c">


As for empty state, I mean the empty state illustration that you see when you are in a new chat. I think that's missing from threads in general, so we could always address that in a separate issue.

jasperhuangg committed:
merge main

jasperhuangg commented:
@shawnborton It's all vertical aligned now!

<img width="386" alt="Screenshot 2023-06-23 at 11 52 43 AM" src="https://github.com/Expensify/App/assets/31285285/f8edc0ef-9d93-4af1-ad56-d310fe5bae21">

jasperhuangg committed:
use shouldShowHorizontalRule

jasperhuangg commented:
> As for empty state, I mean the empty state illustration that you see when you are in a new chat. I think that's missing from threads in general, so we could always address that in a separate issue.

I'll think it's better if we address this in a followup issue since it touches so many other things. I can create a follow up issue.

shawnborton commented:
Nice, looks great! How big is the checkbox? Can you show screenshots of it checked as well?

jasperhuangg commented:
@shawnborton here's it checked/unchecked

<img width="382" alt="Screenshot 2023-06-23 at 12 12 07 PM" src="https://github.com/Expensify/App/assets/31285285/b7880a97-2902-4f92-993b-e0bab7173d99">
<img width="382" alt="Screenshot 2023-06-23 at 12 12 02 PM" src="https://github.com/Expensify/App/assets/31285285/d51241a4-d326-420e-b031-a8e6bd36afee">

shawnborton commented:
Nice. Can you confirm the size of the carets, too? They should be 20x20 but just want to double check.

jasperhuangg commented:
@shawnborton Good looks, seems like they were 14px. I increased their size to 20px, but it didn't look right, and then checked the figma and I think they're supposed to be 16px.

#### 20px
<img width="96" alt="Screenshot 2023-06-23 at 12 23 02 PM" src="https://github.com/Expensify/App/assets/31285285/19b5a4a9-7bb6-4d29-9923-3c3cb7a97302">


#### 16px
<img width="99" alt="Screenshot 2023-06-23 at 12 23 13 PM" src="https://github.com/Expensify/App/assets/31285285/2d02c2b6-8561-4cf7-a65a-ff38ba921c57">

jasperhuangg commented:
@fedirjh @thienlnam this is ready for review!

fedirjh reviewed (commented):
Left few comments

jasperhuangg committed:
push missing changes

jasperhuangg committed:
style

jasperhuangg committed:
remove unused param

jasperhuangg committed:
add/use styles.backgroundBG

jasperhuangg committed:
move maskAsDone and markAsIncomplete translations to task

jasperhuangg commented on src/components/ReportActionItem/TaskView.js:
```diff
@@ -0,0 +1,124 @@
+            <MenuItem
+                label={props.translate('task.assignee')}
+                title={ReportUtils.getDisplayNameForParticipant(props.report.managerID)}
+                icon={UserUtils.getAvatar(
```
Ah I think you may need a Web-E PR to handle this, I thought it was sorted on `main` so elected not to push the PR but seems it wasn't

jasperhuangg replied:
Holding on https://github.com/Expensify/Web-Expensify/pull/37967, which should fix this

jasperhuangg committed:
move newassigneeerror to task

jasperhuangg committed:
resolve merge conflicts

jasperhuangg commented:
> Also when the title is short the arrow does not align to the top.
> 
> <img alt="Screenshot 2023-06-24 at 8 38 59 PM" width="574" src="https://user-images.githubusercontent.com/36869046/248509428-8293c5ef-15a6-49fd-9d04-a1ff9f1f35c0.png">


@fedirjh I think everything is aligned how it's supposed to in this particular case:

<img width="1013" alt="Screenshot 2023-06-27 at 11 09 35 AM" src="https://github.com/Expensify/App/assets/31285285/2982cfbd-1fcd-40c0-8311-7c6363266421">


Here's how the alignment looks in the Figma file. You can imagine that if the second line didn't exist then the alignment would be the same. The checkmark is intended to be vertically center-aligned with the first line of text.
<img width="374" alt="Screenshot 2023-06-27 at 11 10 22 AM" src="https://github.com/Expensify/App/assets/31285285/f75a94bc-1ff8-4f15-96c9-4c6d81135163">

jasperhuangg committed:
fix workspace menu item avatar display

jasperhuangg committed:
fix warning

jasperhuangg commented on src/components/ReportActionItem/TaskView.js:
```diff
@@ -0,0 +1,124 @@
+                })}
+                style={({hovered, pressed}) => [styles.ph5, styles.pv2, StyleUtils.getButtonBackgroundColorStyle(getButtonState(hovered, pressed), true)]}
+                ref={props.forwardedRef}
+                disabled={!isOpen}
```
fixed!

jasperhuangg commented:
> We have another crash
> 
> <img alt="Screenshot 2023-06-24 at 8 32 38 PM" width="1440" src="https://user-images.githubusercontent.com/36869046/248508430-996f2416-d6db-4bd4-aa07-220d33f91478.png">

fixed this!

jasperhuangg commented:
@fedirjh good for another look! Thanks again in advance for your review :)

jasperhuangg commented:
> > I think everything is aligned how it's supposed to in this particular case:
> 
> @jasperhuangg check the video , hover over the task title, it's noticeable that that the arrow is not centered, on the other hand the assignee arrow is centered
> 
>  Screen.Recording.2023-06-27.at.7.39.21.PM.mov

@shawnborton what do you think about this? I thought the arrow was supposed to be vertically center aligned with the text/checkbox. And since they aren't vertically center aligned with the entire component, then the arrow won't be either.

jasperhuangg committed:
resolve conflicts with main

jasperhuangg committed:
resolve conflicts

jasperhuangg committed:
ensure horizontal rule refreshes correctly

fedirjh reviewed (commented):
LHN doesn't display latest actions , mark as done, reopen task, task assigned to someone else 

https://github.com/Expensify/App/assets/36869046/16d7b61a-fc2b-44c8-9cfd-08cd5875a8d8

jasperhuangg committed:
fix case with empty assignee

jasperhuangg committed:
style

jasperhuangg committed:
fix translations

jasperhuangg committed:
remove unused style

jasperhuangg committed:
use utility function

jasperhuangg committed:
add accessibility label

jasperhuangg committed:
merge main

jasperhuangg committed:
fix merge conflicts

jasperhuangg commented on src/pages/tasks/TaskDescriptionPage.js:
I'm unsure, but we don't need to address that now. That can be part of a cleanup issue later on.

jasperhuangg committed:
fix task header background color

jasperhuangg committed:
allow description to occupy multiple lines

jasperhuangg committed:
fix inverted elements on ios

jasperhuangg committed:
Update src/styles/styles.js

Co-authored-by: Fedi Rajhi <fedirjh@gmail.com>

jasperhuangg committed:
Merge branch 'main' of github.com:Expensify/App into jasper-taskRedesign

jasperhuangg committed:
Merge branch 'jasper-taskRedesign' of github.com:Expensify/App into jasper-taskRedesign

jasperhuangg committed:
fix warnings

jasperhuangg committed:
fix background color for task headers

fedirjh reviewed (commented):
Few comment

jasperhuangg commented on src/pages/home/ReportScreen.js:
```diff
@@ -293,11 +293,14 @@ class ReportScreen extends React.Component {
-                                    personalDetails={this.props.personalDetails}
-                                />
+                            {ReportUtils.isTaskReport(this.props.report) && this.props.isSmallScreenWidth && (
+                                <View style={styles.borderBottom}>
+                                    <View style={[styles.darkBG, styles.pl0]}>
+                                        <View style={[styles.ph5, styles.pb3]}>
+                                            <TaskHeaderActionButton report={this.props.report} />
+                                        </View>
+                                    </View>
+                                </View>
```
yes, good catch. this was copied over from a different file, I'll change it!

jasperhuangg replied:
See https://github.com/Expensify/App/pull/20841#discussion_r1253674721 for more

jasperhuangg commented on src/pages/home/ReportScreen.js:
```diff
@@ -293,11 +293,14 @@ class ReportScreen extends React.Component {
-                                />
+                            {ReportUtils.isTaskReport(this.props.report) && this.props.isSmallScreenWidth && (
+                                <View style={styles.borderBottom}>
+                                    <View style={[styles.darkBG, styles.pl0]}>
```
hmmm interesting, this never happened for me

jasperhuangg replied:
you're on an iOS sim? it appears to render fine for me
<img width="439" alt="Screenshot 2023-07-05 at 2 27 23 PM" src="https://github.com/Expensify/App/assets/31285285/8f7348d0-66e6-4ace-8348-253816ca4753">

jasperhuangg replied:
ahh interesting, that only happens when you combine the views that you're talking about [here](https://github.com/Expensify/App/pull/20841/files#r1253572989)

jasperhuangg replied:
fixed

jasperhuangg committed:
fix review comments

jasperhuangg committed:
typo

jasperhuangg committed:
Update src/components/TaskHeaderActionButton.js

Co-authored-by: Fedi Rajhi <fedirjh@gmail.com>

jasperhuangg committed:
fix merge conflicts

jasperhuangg committed:
import const

jasperhuangg committed:
add missing border bottom styles to task headergis

jasperhuangg committed:
remove unused

jasperhuangg committed:
fix duplicated border bottom styles, fix display bug with incorrect bg color

jasperhuangg committed:
remove unused

jasperhuangg committed:
remove unused

jasperhuangg committed:
disable picking assignee via header for closed tasks

jasperhuangg committed:
disable checkbox for canceled tasks

jasperhuangg committed:
style

thienlnam commented:
Appears to have a few conflicts with the latest header updates

jasperhuangg committed:
fix merge conflicts

jasperhuangg committed:
fix merge conflicts

jasperhuangg committed:
remove unused

jasperhuangg committed:
fix crash from merge

jasperhuangg committed:
fix merge conflicts with main

jasperhuangg committed:
disable title menu for canceled tasks

jasperhuangg committed:
disable title menu for canceled tasks

jasperhuangg committed:
disable title menu for canceled tasks

thienlnam commented:
cc @fedirjh When you get a chance could you review today? Would love to get this out - thank you!

jasperhuangg commented:
@thienlnam we've been chatting in the Slack DM working through the review! He's completing the reviewer checklist as we speak.

jasperhuangg committed:
linter

thienlnam commented:
Oh lovely - thank you guys!

jasperhuangg committed:
fix conflicts

jasperhuangg committed:
fix conflicts

fedirjh reviewed (approved):
LGTM and tests well 🚀

thienlnam commented on src/components/MenuItem.js:
```diff
@@ -63,10 +63,13 @@ const defaultProps = {
     furtherDetails: '',
     furtherDetailsIcon: undefined,
+    isSmallAvatarSubscriptMenu: false,
+    title: '',
```
I don't see title used here

jasperhuangg replied:
it's used below

thienlnam reviewed (commented):
Code looking solid! Just going to pull and test a bit

thienlnam commented:
<img width="1135" alt="Screenshot 2023-07-06 at 4 31 36 PM" src="https://github.com/Expensify/App/assets/30609178/b8a03c97-0249-4e45-ad69-cb19805e010f">

Noticed this in testing - when it's completed you can still highlight over the title which looks odd

thienlnam commented:
But other than that it tests great and looks awesome - nice work!

thienlnam commented:
Also while you're here - could you please update the Cancelled Task parameter optimisticReportActionID to cancelledTaskReportActionID

jasperhuangg commented:
> <img width="1135" alt="Screenshot 2023-07-06 at 4 31 36 PM" src="https://github.com/Expensify/App/assets/30609178/b8a03c97-0249-4e45-ad69-cb19805e010f">
> 
> 
> 
> Noticed this in testing - when it's completed you can still highlight over the title which looks odd
> 
> 

Ah hmmm we fixed this, maybe it wasn't pushed, lemme check

jasperhuangg committed:
disable hovered/pressed styles for task title menu item for completed tasks

jasperhuangg committed:
apply disabled styles to title text for completed task

jasperhuangg committed:
pass cancelledTaskReportActionID

thienlnam commented on src/libs/actions/Task.js:
```diff
@@ -624,11 +625,7 @@ function cancelTask(taskReportID, taskTitle, originalStateNum, originalStatusNum
-
-function isTaskCanceled(taskReport) {
-    return taskReport.stateNum === CONST.REPORT.STATE_NUM.SUBMITTED && taskReport.statusNum === CONST.REPORT.STATUS.CLOSED;
+    API.write('CancelTask', {, cancelledTaskReportActionID: optimisticReportActionID, taskReportID}, {optimisticData, failureData});
```
Shouldn't have the comma here

jasperhuangg replied:
I'm so confused, that comma isn't there for me locally 😕

shawnborton commented:
@thienlnam how did you get this screenshot where the description and assignee are gray?
![image](https://github.com/Expensify/App/assets/2319350/870a5b59-38cf-482e-b25e-6481178493d6)


Also, let me know when we have final final screenshots for testing. Perhaps we can create a test build too?

shawnborton commented:
I'm a little confused by this:
<img width="838" alt="image" src="https://github.com/Expensify/App/assets/2319350/3bf0d7f5-1f8c-42a7-b2f1-032ae284d19e">


If you click the checkbox to complete a task, it should be the same as clicking the green button up top. 

And then, why do things gray out when the task is complete?

jasperhuangg committed:
Update Task.js

jasperhuangg committed:
merge main

jasperhuangg committed:
don't grey out menu items when canceled/closed

jasperhuangg committed:
fix merge conflicts

jasperhuangg commented:
@shawnborton Here's how it looks when it's canceled:

<img width="1357" alt="Screenshot 2023-07-07 at 11 16 45 AM" src="https://github.com/Expensify/App/assets/31285285/39c7a156-8e25-4fed-8d54-12c09f8a7e8c">

And here's how it looks when it's disabled:
<img width="1356" alt="Screenshot 2023-07-07 at 11 16 54 AM" src="https://github.com/Expensify/App/assets/31285285/11677449-d25a-4c57-8213-e46e631b0934">

shawnborton commented:
Cool, much better. So then why do we have a green button in the header when the checkbox is already checked?

jasperhuangg commented:
> Cool, much better. So then why do we have a green button in the header when the checkbox is already checked?

Ah hmm didn't realize that wasn't supposed to be there. The text changes to "Mark as incomplete" when the checkbox is checked. But I can remove it.

jasperhuangg commented:
<img width="1353" alt="Screenshot 2023-07-07 at 11 26 16 AM" src="https://github.com/Expensify/App/assets/31285285/dcaa563e-049b-46d0-8749-23da463e5e3e">

jasperhuangg committed:
prettier

shawnborton commented:
Yeah, my understanding is that hitting the checkbox or hitting the green button does the same thing.

jasperhuangg commented:
> Yeah, my understanding is that hitting the checkbox or hitting the green button does the same thing.

What do you mean by that? In how it was before, they would do the same thing. Checking the checkbox changes the button text from "mark as done" to "mark as incomplete". Unchecking the checkbox changes it back. And clicking the button does vice versa.

shawnborton commented:
Got it, then I think we're on the same page. But the idea is that the checkbox basically does the same thing as the header buttons do.

thienlnam reviewed (dismissed):
Awesome, thank you!

jasperhuangg committed:
resolve conflicts with main

thienlnam reviewed (approved):
Merging this quickly with haste since it's a big PR that continues to get merge conflicts

Design comments have been addressed

OSBotify commented:
:hand: This PR was not deployed to staging yet because QA is ongoing. It will be automatically deployed to staging after the next production release.

ayazhussain79 commented:
App crashes when creating task  https://expensify.slack.com/archives/C049HHMV9SM/p1688766194182809

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5514102677) to staging by https://github.com/thienlnam in version: 1.3.39-0 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

OSBotify commented:
🚀 [Deployed](https://github.com/Expensify/App/actions/runs/5538073623) to production by https://github.com/francoisl in version: 1.3.39-11 🚀

 platform | result 
 ---|--- 
🤖 android 🤖|success ✅ 
🖥 desktop 🖥|success ✅
🍎 iOS 🍎|success ✅ 
🕸 web 🕸|success ✅

s77rt commented on src/components/ReportActionItem/TaskView.js:
```diff
@@ -0,0 +1,148 @@
+                <MenuItem
+                    label={props.translate('task.assignee')}
+                    title={ReportUtils.getDisplayNameForParticipant(props.report.managerID)}
+                    icon={UserUtils.getAvatar(
+                        PersonalDetailsUtils.getPersonalDetailsByIDs([props.report.managerID], props.currentUserPersonalDetails.accountID)[0].avatar,
+                        props.report.managerID,
+                    )}
```
This caused a regression https://github.com/Expensify/App/issues/22657. `PersonalDetailsUtils.getPersonalDetailsByIDs` could return an empty array e.g. for users that you have not interacted with before (no personaldetails). Then we will be trying to access property `avatar` of `undefined`.

StevenKKC commented on src/pages/home/report/ReportActionItem.js:
```diff
@@ -541,6 +554,17 @@ export default compose(
+            prevProps.report.managerID === nextProps.report.managerID &&
+            prevProps.report.managerEmail === nextProps.report.managerEmail &&
+            prevProps.isOnlyReportAction === nextProps.isOnlyReportAction,
     ),
```
Hi ✋ Coming from https://github.com/Expensify/App/issues/24267

We should add comparing of `report.amount` to dynamically update total amount in money request view.

jasperhuangg replied:
Hmm that doesn't relate to tasks, which this PR implements, but good thinking! That seems like that would solve the problem.

situchan commented on src/components/ReportActionItem/TaskView.js:
```diff
@@ -0,0 +1,148 @@
+
+    return (
+        <View>
+            <PressableWithSecondaryInteraction
```
What was reason for using `PressableWithSecondaryInteraction` in `TaskView` instead of just `PressableWithFeedback`?
I don't see any secondary interaction.
